### PR TITLE
fix Expected both nodes to be constants error

### PR DESCRIPTION
**Pull request by evykassirer**
[asana task](https://app.asana.com/0/201972293666219/228523808431564)

and added the fix to https://sentry.io/socratic/athena/issues/193077935/ because it was also an equations thing

### CODE REVIEW

File: lib/solveEquation.js [hunk 1]
```diff
@@ -225,9 +232,15 @@ function solveConstantEquation(equation, debug, steps=[]) {
       lastStep.asciimath, equation.comparator);
   }
 
+  // If the left or right side didn't have any steps, unecessary parens
```
Comment (aelnaiem) [inline lib/solveEquation.js]: unnecessary*

File: test/solveEquation.test.js [hunk 2]
```diff
@@ -107,11 +107,35 @@ describe('constant comparison support', function () {
...
+  it(equationString + ' throws error', function () {
+    assert.throws(() => solveEquation(leftNode, rightNode, comparator, debug),
+                  errorMessage);
+  });
+}
+
+describe('solveEquation errors', function() {
+  const tests = [
+    ['( x + 2) ^ ( 2) - x ^ ( 2) = 4( x + 1)', '=',
+      'Math error: Potential infinite loop for equation ' +
```
Comment (aelnaiem) [inline test/solveEquation.test.js]: Hahaha this is a very specific error test

**Comment by aelnaiem** [comment]
Couple comments, otherwise 👍

File: lib/solveEquation.js (hunk 1 above)
Comment (evykassirer) [inline lib/solveEquation.js]: god damn :p

File: test/solveEquation.test.js (hunk 2 above)
Comment (evykassirer) [inline test/solveEquation.test.js]: yeah apparently you have to test the exact string... :/ maybe I'll look into it more tomorrow

**Pull request merged**